Tools/LyTestTools/ly_test_tools/cli/codeowners_hint.py

```diff
@@ -0,0 +1,232 @@
+
+
+def _pretty_print_failure(print_fn, found_codeowners_path, matched_path, original_target,
+                          default_alias=_DEFAULT_CODEOWNER_ALIAS):
+    """
+    :param print_fn: function to call when logging strings
+    :param found_codeowners_path: verified path to a GitHub CODEOWNERS file which, empty when missing
+    :param matched_path: entry matched in the CODEOWNERS file, empty when not matched
+    :param original_target: the path which matching was attempted on
+    :param default_alias: who to contact as no owner was found
+    """
+    if not found_codeowners_path:
+        print_fn(f"No GitHub CODEOWNERS file was found for '{original_target}'")
+    else:
+        if not matched_path:
+            print_fn(f"No ownership information for '{original_target}' found in file {found_codeowners_path}")
+        else:
+            print_fn(f"Ownership for '{matched_path}' is explicitly empty in file {found_codeowners_path}")
+    print_fn(f"For additional support please reach out to: {default_alias}")
+
+
+def _main() -> int:
+    parser = argparse.ArgumentParser(description="Display GitHub CODEOWNERS information for a path to stdout")
+    parser.add_argument('target', metavar='T',
+                        help="file path to find an owner for")
+    parser.add_argument('-c', '--codeowners',
+                        help="path to a GitHub CODEOWNERS file, when not set it will be searched for in the repo "
+                             "containing the target")
+    parser.add_argument('-d', '--default_alias', default=_DEFAULT_CODEOWNER_ALIAS,
+                        help="location to reach out for support when ownership cannot be determined")
+    parser.add_argument('-p', '--pretty_print', action='store_true',
+                        help="output ownership info as a friendly message instead of only alias(es)")
+    parser.add_argument('-s', '--silent', action='store_true',
+                        help="Suppress warning messages and only print ownership information")
+    args = parser.parse_args()
+
+    if args.silent:
+        logging.disable()
+    else:
+        logging.basicConfig()
+
+    if args.codeowners:
+        matched_path, owner_aliases = get_codeowners_from(args.target, args.codeowners)
+        found_codeowners = os.path.isfile(args.codeowners)
+    else:
+        matched_path, owner_aliases, found_codeowners = get_codeowners(args.target)
+
+    if owner_aliases and matched_path and found_codeowners:
+        if args.pretty_print:
+            _pretty_print_success(print, found_codeowners, matched_path, owner_aliases)
+        else:
+            print(owner_aliases)
+        return 0
+    else:
+        if args.pretty_print:
+            _pretty_print_failure(print, found_codeowners, matched_path, args.target, args.default_alias)
+        else:
+            print(args.default_alias)
+        return 1
+
+    logger.error("Unexpected abnormal exit")
+    return -1
+
+
+if __name__ == '__main__':
```

Comment:
Having this script entrypoint exist buried in the /Tools/ folder avoids some of the user accessibility of a named file existing in the /scripts/ folder. I'm going to propose diff with this main-hook in a separate file, and check back on review comments.